lib/src/file_util.rs

```diff
@@ -134,6 +148,47 @@ pub fn persist_content_addressed_temp_file<P: AsRef<Path>>(
     }
 }
 
+/// Symlinks are always available on UNIX, but on Windows they require admin or
+/// Developer Mode
+#[cfg(windows)]
+pub fn try_symlink<P: AsRef<Path>, Q: AsRef<Path>>(
+    original: P,
+    link: Q,
+    is_dir: bool,
+) -> Result<(), SymlinkError> {
+    const ERROR_PRIVILEGE_NOT_HELD: i32 = 1314;
+
+    // test whether the symlink already exists as the wrong *kind* of
+    // link (directory vs file); if so we have to recreate it
+    if is_dir && matches!(symlink_metadata(link.as_ref()), Ok(meta) if !meta.is_dir()) {
+        remove_dir(link.as_ref()).map_err(SymlinkError::InternalError)?;
+    }
```

Comment:
What if `!is_dir` and directory symlink already exists?

(I don't know if symlink can be overwritten on Windows, so ignore me if that just works.)

Reply:
It can be overwritten, as long as you're overwriting a link-to-directory with a link-to-directory (and so on for files).

Reply:
then we'll need to handle link-to-dir -> link-to-file case, too?

ah, btw, `symlink()` on Unix fails if the entry already exists. The existing entry should have been deleted by TreeState or LocalWorkingCopy, so we wouldn't have to do `remove_dir()` or `remove_file()`.